From 14814914e84a8ddc743fe9224de6a43689b92889 Mon Sep 17 00:00:00 2001
From: Alex Shevelkin <ash@tvip.ru>
Date: Tue, 16 Feb 2021 16:53:38 +0300
Subject: [PATCH] Allow 'iw reg set XX' to work

---
 drivers/net/wireless/rtl88x2cs/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/rtl88x2cs/Makefile b/drivers/net/wireless/rtl88x2cs/Makefile
index 546ca1925..814cf1e1b 100755
--- a/drivers/net/wireless/rtl88x2cs/Makefile
+++ b/drivers/net/wireless/rtl88x2cs/Makefile
@@ -2317,6 +2317,7 @@ endif
 ########### HAL_RTL8822C #################################
 ifeq ($(CONFIG_RTL8822C), y)
 include $(TopDIR)/drivers/net/wireless/rtl88x2cs/rtl8822c.mk
+EXTRA_CFLAGS += -DCONFIG_RTW_IOCTL_SET_COUNTRY
 endif
 
 ########### HAL_RTL8814B #################################
-- 
2.30.0

