From ffb8745c6ce08eb053167a98eff50599e21f0846 Mon Sep 17 00:00:00 2001
From: Alex Shevelkin <ash@tvip.ru>
Date: Sat, 20 Feb 2021 15:13:11 +0300
Subject: [PATCH] cfg80211: regulatory: set AU as default country code

---
 net/wireless/reg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/wireless/reg.c b/net/wireless/reg.c
index 763a45655ac2..d655f65bd7b8 100644
--- a/net/wireless/reg.c
+++ b/net/wireless/reg.c
@@ -262,7 +262,7 @@ static const struct ieee80211_regdomain world_regdom = {
 static const struct ieee80211_regdomain *cfg80211_world_regdom =
 	&world_regdom;
 
-static char *ieee80211_regdom = "00";
+static char *ieee80211_regdom = "AU";
 static char user_alpha2[2];
 static const struct ieee80211_regdomain *cfg80211_user_regdom;
 
-- 
2.30.0

