From aa4192e5b9c6cefa36f56294fac822b25ebc7030 Mon Sep 17 00:00:00 2001
From: Vyacheslav Bocharov <devel@lexina.in>
Date: Thu, 10 Jun 2021 12:45:34 +0300
Subject: [PATCH 06/11] fix uuid for partitions

---
 lib/uuid.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/lib/uuid.c b/lib/uuid.c
index 5bc68674d02..09f614ad696 100644
--- a/lib/uuid.c
+++ b/lib/uuid.c
@@ -87,6 +87,7 @@ int uuid_str_valid(const char *uuid)
 }
 
 #ifdef CONFIG_PARTITION_TYPE_GUID
+
 static const struct {
 	const char *string;
 	efi_guid_t guid;
@@ -100,6 +101,25 @@ static const struct {
 	{"swap",	PARTITION_LINUX_SWAP_GUID},
 	{"lvm",		PARTITION_LINUX_LVM_GUID},
 	{"u-boot-env",	PARTITION_U_BOOT_ENVIRONMENT},
+#ifdef CONFIG_AML_GPT
+		{"bootloader",	PARTITION_ANDROID_BOOTLOADER_GUID},
+		{"bootloader2",	PARTITION_ANDROID_BOOTLOADER2_GUID},
+		{"recovery",	PARTITION_ANDROID_RECOVERY_GUID},
+		{"misc",		PARTITION_ANDROID_MISC_GUID},
+		{"metadata",	PARTITION_ANDROID_METADATA_GUID},
+		{"system",		PARTITION_ANDROID_SYSTEM_GUID},
+		{"cache",		PARTITION_ANDROID_CACHE_GUID},
+		{"data",		PARTITION_ANDROID_DATA_GUID},
+		{"persistent",	PARTITION_ANDROID_PERSISTENT_GUID},
+		{"vendor",		PARTITION_ANDROID_VENDOR_GUID},
+		{"config",		PARTITION_ANDROID_CONFIG_GUID},
+		{"factory",		PARTITION_ANDROID_FACTORY_GUID},
+		{"factory_alt",	PARTITION_ANDROID_FACTORY_ALT_GUID},
+		{"fastboot",	PARTITION_ANDROID_FASTBOOT_GUID},
+		{"tertiary",	PARTITION_ANDROID_TERTIARY_GUID},
+		{"oem",			PARTITION_ANDROID_OEM_GUID},
+		{"default",		PARTITION_LINUX_DEFAULT_GUID},
+#endif
 };
 
 /*
-- 
2.30.2

